projects/common/mtl_renderer.cpp

```diff
@@ -911,6 +919,184 @@ NS::Error* CreateGraphicsPipeline1(
     return pError;
 }
 
+NS::Error* CreateGraphicsPipeline2(
+    MetalRenderer*            pRenderer,
+    MetalShader*              pVsShaderModule,
+    MetalShader*              pFsShaderModule,
+    MTL::PixelFormat          rtvFormat,
+    MTL::PixelFormat          dsvFormat,
+    MetalPipelineRenderState* pPipelineRenderState,
+    MetalDepthStencilState*   pDepthStencilState,
+    uint32_t*                 pOptionalVertexStrides)
+{
+    MTL::VertexDescriptor* pVertexDescriptor = MTL::VertexDescriptor::alloc()->init();
+    if (pVertexDescriptor != nullptr) {
+        const uint32_t    inputCount          = 4;
+
+        MTL::VertexFormat formats[inputCount] = {
+            MTL::VertexFormatFloat3,  // Position
+            MTL::VertexFormatFloat2,  // TexCoord
+            MTL::VertexFormatFloat3,  // Normal
+            MTL::VertexFormatFloat3}; // Tangent
+
+        uint32_t strides[inputCount] = {12, 8, 12, 12};
```

Comment:
It would be coo if there was like a size functino for the metal formats so we can use it here instead of hard coding it.

Reply:
Added BitsPerElement() function